cmd/cli/devstack/devstack.go

```diff
@@ -206,12 +207,39 @@ func runDevstack(cmd *cobra.Command, cfg types.BacalhauConfig, fsr *repo.FsRepo,
 		devstack.WithComputeConfig(computeConfig),
 		devstack.WithRequesterConfig(requesterConfig),
 	)
+	userKeyPath, err := fsr.UserKeyPath()
+	if err != nil {
+		return err
+	}
+	userKey, err := baccrypto.LoadUserKey(userKeyPath)
+	if err != nil {
+		return err
+	}
+	executionDir, err := fsr.ExecutionDir()
+	if err != nil {
+		return err
+	}
 	if IsNoop {
 		options = append(options, devstack.WithDependencyInjector(devstack.NewNoopNodeDependencyInjector()))
 	} else if ODs.ExecutorPlugins {
-		options = append(options, devstack.WithDependencyInjector(node.NewExecutorPluginNodeDependencyInjector(cfg)))
+		pluginPath, err := fsr.EnginePluginsDir()
+		if err != nil {
+			return err
+		}
+		options = append(options, devstack.WithDependencyInjector(
+			node.NewExecutorPluginNodeDependencyInjector(
+				cfg,
+				pluginPath,
+				executionDir,
+				userKey,
+			)))
 	} else {
-		options = append(options, devstack.WithDependencyInjector(node.NewStandardNodeDependencyInjector(cfg)))
+		options = append(options, devstack.WithDependencyInjector(
+			node.NewStandardNodeDependencyInjector(
+				cfg,
+				executionDir,
```

Comment:
This is where things can get confusing. we should just pass the config, and expose `cfg.ExecutionDir()`

Reply:
Similar question as well: what creates the ExecutionDir? Asking since the executionDir provided here must exist on the filesystem, `NewStandardNodeDependencyInjector` (and many other constructors) expect the paths they are handed to exist on disk, if the path doesn't exist they error.

Reply:
If it makes things easier, create the dirs and their parent path the first time you call ExecutionDir() if they don't exist, assuming we use the dir when we call this method and we don't call it for requester nodes as well and end up creating unnecessary dirs.

Otherwise we might need another explicit GetOrCreatExecutionDir() method

Reply:
> create the dirs and their parent path the first time you call ExecutionDir() if they don't exist

This is what `r.ExecutionDir()` does, it either creates the directory if it doesn't exist, or returns its path if it does.